frontend/src/views/Join.tsx

```diff
@@ -1,16 +1,27 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { PrimaryButton } from '../components/core/Button';
-import { connect, disconnect } from '../api/Socket';
+import Input from '../components/core/Input';
+import { connect, disconnect, sendUser } from '../api/Socket';
+
+// Create constants for the subscription and send message URLs.
+const SOCKET_ENDPOINT:string = '/api/v1/socket/join-room-endpoint';
 
 function JoinGamePage() {
+  // Declare nickname state variable.
+  const [nickname, setNickname] = useState('');
+
   return (
     <div>
-      <PrimaryButton onClick={() => connect('/join-room-endpoint')}>
+      <Input placeholder="Enter your nickname" onChange={(event) => setNickname(event.target.value)} />
+      <PrimaryButton onClick={() => connect(SOCKET_ENDPOINT, nickname)}>
         Connect
       </PrimaryButton>
       <PrimaryButton onClick={disconnect}>
         Disconnect
       </PrimaryButton>
+      <PrimaryButton onClick={() => sendUser(nickname)}>
+        Send User Information
```

Comment:
This all looks great on the UI 👍

Reply:
Thank you! I'll be removing it in the next PR, but much appreciated 👍